### PR TITLE
Reaper : Make the pause_deletion expiration time configurable : Close…

### LINKED ISSUE #4138: Make the pause_deletion expiration time configurable
Motivation
----------
The `pause_deletion` expiration time in memcached is currently hardcoded

Modification
------------
Make the pause_deletion expiration time configurable

### PR DESCRIPTION
**Pull request by cserf**
Reaper : Make the pause_deletion expiration time configurable : Close #4138

### CODE REVIEW

File: lib/rucio/daemons/reaper/reaper2.py [hunk 1]
```diff
@@ -240,6 +240,13 @@ def get_rses_to_hostname_mapping():
 
     result = REGION.get('rse_hostname_mapping')
     if result is NO_VALUE:
+        semaphore = REGION.get('semaphore')
```
Comment (mlassnig) [inline lib/rucio/daemons/reaper/reaper2.py:243]: i don't think this is correct. multiple reapers on the same node could set/delete this value. the REGION.get() needs to atomically wait, not check it with an `if` later. or am i misunderstanding what you're trying to accomplish here?

File: lib/rucio/daemons/reaper/reaper2.py [hunk 2]
```diff
@@ -240,6 +240,13 @@ def get_rses_to_hostname_mapping():
...
+        if semaphore is NO_VALUE:
```
Comment (bari12) [inline lib/rucio/daemons/reaper/reaper2.py:244]: The semaphore should be removed. I don't think it has the wished effect and will not do much.

File: lib/rucio/daemons/reaper/reaper2.py [hunk 3]
```diff
@@ -436,6 +444,9 @@ def reaper(rses, include_rses, exclude_rses, vos=None, chunk_size=100, once=Fals
             GRACEFUL_STOP.wait(30)
             continue
         start_time = time.time()
+        memcached_expiration_time = get('reaper', 'memcached_expiration_time')
```
Comment (bari12) [inline lib/rucio/daemons/reaper/reaper2.py:447]: Let's call this `pause_deletion_time` or so. It has nothing todo with memcache per se, that's just the mechanism how it is implemented.

File: lib/rucio/daemons/reaper/reaper2.py [hunk 4]
```diff
@@ -482,14 +494,16 @@ def reaper(rses, include_rses, exclude_rses, vos=None, chunk_size=100, once=Fals
...
-                result = REGION.get('pause_deletion_%s' % rse_id, expiration_time=120)
+                result = REGION.get('pause_deletion_%s' % rse_id, expiration_time=memcached_expiration_time)
                 if result is not NO_VALUE:
-                    logging.info('%s Not enough replicas to delete on %s during the previous cycle. Deletion paused for a while', prepend_str, rse_name)
                     continue
                 logging.debug('%s Working on %s. Percentage of the total space needed %.2f', prepend_str, rse_name, needed_free_space / tot_needed_free_space * 100)
                 rse_hostname, rse_info = rses_hostname_mapping[rse_id]
                 rse_hostname_key = '%s,%s' % (rse_id, rse_hostname)
-                payload_cnt = list_payload_counts(executable, older_than=600, hash_executable=None, session=None)
+                payload_cnt = REGION.get('payload_cnt', expiration_time=30)
```
Comment (bari12) [inline lib/rucio/daemons/reaper/reaper2.py:503]: I would build this into the `list_payload_counts` directly.

**Review by bari12 — CHANGES_REQUESTED** [review]
Hi @cserf 
Sorry for the long delay on this. A few comments.

**Comment by bari12** [comment]
Closing, most of this is solved by the PR @tbeerman did already. If needed we can add a few missing parts still.

**Pull request closed without merging**